stdlib/fractions.pyi

```diff
@@ -20,11 +20,23 @@ else:
     @overload
     def gcd(a: Integral, b: Integral) -> Integral: ...
 
+_IR = TypeVar("_IR", covariant=True)
+
+class _ConvertibleToIntegerRatio(Protocol[_IR]):
+    def as_integer_ratio(self) -> _IR: ...
+
 class Fraction(Rational):
     @overload
     def __new__(cls, numerator: int | Rational = 0, denominator: int | Rational | None = None) -> Self: ...
     @overload
     def __new__(cls, value: float | Decimal | str, /) -> Self: ...
+
+    if sys.version_info >= (3, 14):
+        @overload
+        def __new__(cls, value: _ConvertibleToIntegerRatio[tuple[int | Rational, int | Rational]]) -> Self: ...
+        @overload
+        def __new__(cls, value: _ConvertibleToIntegerRatio[Iterable[int | Rational]]) -> Self: ...
```

Comment:
Doesn't this overload make the one immediately above it redundant? `tuple[int | Rational, int | Rational]` is a subtype of `Iterable[int | Rational]`

I feel like I'd be okay only including the `tuple` overload, not this one. Even if `Fraction` accepts objects that return non-tuple iterables from their `as_integer_ratio()` methods, it doesn't seem like a good idea for users to annotate their `as_integer_ratio()` methods as returning non-tuples. The creation of the `Fraction` object will fail if the iterable returned by the method is not of length 2, and "the returned iterable will be of length 2" can't be expressed if the method is annotated as returning a non-tuple.